test/distributed/test_store.py

```diff
@@ -130,6 +130,23 @@ def _create_store(self):
         store.set_timeout(timedelta(seconds=300))
         return store
 
+    def test_init_pg_and_rpc_with_same_file(self):
+        # Init RPC using file
+        rpc_backend_options = rpc.TensorPipeRpcBackendOptions()
+        rpc_backend_options.init_method = f"file://{self.file.name}"
+        rpc.init_rpc("worker", rank=0, world_size=1, rpc_backend_options=rpc_backend_options)
+
+        # Init PG using file
+        dist.init_process_group("gloo", rank=0, world_size=1, init_method=f"file://{self.file.name}")
+        dist.destroy_process_group()
+        assert os.path.isfile(self.file.name)
+
+        rpc.shutdown()
```

Comment:
Can you also assert that the file is removed after rpc shutdown?

Reply:
``rpc.shutdown()`` does not invoke the destructor of FileStore, so the file will not be removed. I think this maybe a problem in the RPC module.

This could be reproduced in torch 1.12.0 by the following code:

```python
import os
import torch
import torch.distributed.rpc as rpc
import tempfile
print(torch.__version__)


def main():
    file = tempfile.NamedTemporaryFile(delete=False)
    file_name = file.name
    print(f"Using file: {file_name}, File Exists: {os.path.isfile(file_name)}")

    # Init RPC using file
    rpc_backend_options = rpc.TensorPipeRpcBackendOptions()
    rpc_backend_options.init_method = f"file://{file_name}"
    rpc.init_rpc("worker", rank=0, world_size=1, rpc_backend_options=rpc_backend_options)

    rpc.shutdown()
    print(f"After rpc.shutdown, File Exists: {os.path.isfile(file_name)}")


if __name__ == "__main__":
    main()
```

Output:

```
1.12.0+cu113
Using file: /tmp/tmp6we_vxrc, File Exists: True
After rpc.shutdown, File Exists: True
```